mlir/lib/Dialect/Polynomial/IR/PolynomialAttributes.cpp

```diff
@@ -202,11 +202,30 @@ Attribute RingAttr::parse(AsmParser &parser, Type type) {
     polyAttr = attr;
   }
 
+  Polynomial poly = polyAttr.getPolynomial();
+  APInt root(coefficientModulusAttr.getValue().getBitWidth(), 0);
+  bool hasRoot = succeeded(parser.parseOptionalComma());
+  IntegerAttr rootAttr = nullptr;
+  if (hasRoot) {
+    if (failed(parser.parseKeyword("primitiveRoot")))
+      return {};
+
+    if (failed(parser.parseEqual()))
+      return {};
```

Comment:
```suggestion
    if (failed(parser.parseKeyword("primitiveRoot")) ||
        failed(parser.parseEqual()))
      return {};
```

(you can even omit the `failed` and have the same effect, but I personally prefer being explicit.